Source/WebCore/platform/graphics/texmap/coordinated/CoordinatedBackingStoreProxy.cpp

```diff
@@ -363,6 +334,84 @@ void CoordinatedBackingStoreProxy::createOrDestroyTiles(const IntRect& visibleRe
     m_pendingTileCreation = requiredTileCount;
 }
 
+// This is based on the same heuristics used by chromium to compute the tile size.
+IntSize CoordinatedBackingStoreProxy::computeTileSize(int maxTextureSize) const
+{
+    static IntSize overridenTileSize;
+    static std::once_flag onceFlag;
+    std::call_once(onceFlag, [] {
+        const char* tileSizeEnv = getenv("WEBKIT_LAYERS_TILE_SIZE");
+        if (tileSizeEnv && *tileSizeEnv) {
+            auto tokens = String::fromLatin1(tileSizeEnv).split('x');
+            if (!tokens.isEmpty()) {
+                int width = parseInteger<int>(tokens[0]).value_or(0);
+                int height = tokens.size() > 1 ? parseInteger<int>(tokens[1]).value_or(0) : width;
+                overridenTileSize = { width, height };
+            }
+
+            if (overridenTileSize.isEmpty())
+                WTFLogAlways("Invalid value '%s' for WEBKIT_LAYERS_TILE_SIZE, ignoring", tileSizeEnv);
+        }
+    });
+    if (!overridenTileSize.isEmpty())
+        return overridenTileSize;
+
+    IntSize tileSize;
+    if (ProcessCapabilities::canUseAcceleratedBuffers()) {
+        static constexpr int minGPUTileHeight = 256;
+
+        auto gpuTileSize = [&](const IntSize& baseSize) -> IntSize {
+            int width = baseSize.width();
+            int divisor = 4;
+            if (m_contentsRect.width() <= baseSize.width() / 2)
+                divisor = 2;
+            if (m_contentsRect.width() <= baseSize.width() / 4)
+                divisor = 1;
+            int height = roundUpToMultipleOfNonPowerOfTwo(divisor, baseSize.height()) / divisor;
+
+            // Grow default sizes to account for overlapping border texels.
+            width += 2;
+            height += 2;
+
+            // Round GPU default tile sizes to a multiple of 32 to prevent rounding errors.
```

Comment:
Let's see if that solves is ;-) Do you have more insights in the "rounding errors" now?

Reply:
It didn't help